### PR TITLE
Issue945 hugos security considerations text

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

File: draft-ietf-tls-tls13.md [hunk 1]
```diff
@@ -5142,13 +5242,50 @@ to avoid leaking information about the identities due to length.
...
 to be unilaterally or mutually authenticated, this has to be provisioned
 by the application layer. See {{CHHSV17}} for details.
 
+In addition, the analysis of post-handshake authentication from
+[Kraw16] shows that this mechanism serves as a proof of possession of
+the traffic key by the authenticating client - as identified by the
+client's certificate - but does not say when the client acquired the
+knowledge of the traffic key (it could have had it at the time of the
+original handshake or could have learned it at a later point). The
+implication of this fact is that this mechanism does no provide
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: typo: should be "not"

File: draft-ietf-tls-tls13.md [hunk 2]
```diff
@@ -5157,15 +5294,19 @@ server is not able to guarantee full uniqueness of the handshake
...
 does not provide post-compromise security {{CCG16}}, sometimes also referred to
 as backwards or future security. This is in contrast to KCI resistance, which
 describes the security guarantees that a party has after its own long-term
 secret has been compromised.
 
+### External References
+
 The reader should refer to the following references for analysis of the
-TLS handshake {{CHSV16}} {{FGSW16}} {{LXZFH16}}.
+TLS handshake {{CHSV16}} {{CHSV16-2}} {DFGS16}} {{DFGS16}} {{Kraw16}} {{KW16}} {{FGSW16}} {{LXZFH16}}.
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: To match the styling below, you need a colon after "handshake" and a comma after each item

File: draft-ietf-tls-tls13.md [hunk 3]
```diff
@@ -5142,13 +5242,50 @@ to avoid leaking information about the identities due to length.
...
 by the application layer. See {{CHHSV17}} for details.
 
+In addition, the analysis of post-handshake authentication from
+[Kraw16] shows that this mechanism serves as a proof of possession of
+the traffic key by the authenticating client - as identified by the
+client's certificate - but does not say when the client acquired the
+knowledge of the traffic key (it could have had it at the time of the
+original handshake or could have learned it at a later point). The
+implication of this fact is that this mechanism does no provide
+cryptographic attestion for any data exchanged prior to the
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: spelling typo: should be "attestation" instead of "attestion"

File: draft-ietf-tls-tls13.md (hunk 2 above)
Comment (ekr) [inline draft-ietf-tls-tls13.md]: I think I am going to go with colon but no comma.

File: draft-ietf-tls-tls13.md [hunk 4]
```diff
@@ -5142,13 +5242,50 @@ to avoid leaking information about the identities due to length.
...
+long as these are differentiated via the key and/or the labels.
+
+Note that HKDF-Expand implements a pseudorandom function (PRF) with both inputs and
+outputs of variable length. In some of the uses of HKDF in this document
+(e.g., for generating exporters and the resumption_master_secret), it is necessary
+that the application of HKDF-Expand be collision-resistant, namely, it should
+be infeasible to find two different inputs to HKDF-Expand that output the same
+value. This requires the underlying hash function to be collision resistant
+and the output length from HKDF-Expand to be of size at least 256 bits (or as
+much as needed for the hash function to prevent finding collisions).
```
Comment (kaduk) [inline draft-ietf-tls-tls13.md]: Some of this paragraph feels like it's duplicating portions of #security-handshake (the PSK binder bits); should some content get trimmed from there?

**Comment by ekr** [comment]
Merged by hand.

**Pull request closed without merging**